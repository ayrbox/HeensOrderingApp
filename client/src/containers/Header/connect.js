import { connect } from 'react-redux';
import Header from './header';

// TODO: import action to open order modal (that will just toggle open)
// import

import { openOrderModal } from '../../actions/orderActions';

const mapState = state => ({
  testState: state.auth,
});

export default connect(mapState, {
  openOrderModal,
})(Header);
