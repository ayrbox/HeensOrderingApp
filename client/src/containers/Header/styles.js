const styles = theme => ({
  root: {
    flexGrow: 1,
    zIndex: theme.zIndex.drawer + 1,
  },
  toolbar: {
    justifyContent: 'space-between',
  },
  navItemsContainer: {
    display: 'flex',
  },
  navItem: {
    display: 'flex',
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
});

export default styles;
