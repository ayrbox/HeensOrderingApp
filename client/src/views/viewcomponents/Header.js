import React from "react";
import { Link } from "react-router-dom";

const Header = ({ title }) => (
  <nav className="navbar navbar-expand-md navbar-dark fixed-top bg-dark">
    <div className="container">
      <Link to="/" className="navbar-brand">
        {title}
      </Link>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarCollapse"
        aria-controls="navbarCollapse"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navbarCollapse">
        <ul className="navbar-nav mr-auto">
          <li className="nav-item active">
            <Link className="nav-link" to="/">
              Home <span className="sr-only">(current)</span>
            </Link>
          </li>
          <li className="nav-item">
            <Link className="nav-link" to="/">
              Link
            </Link>
          </li>
          <li className="nav-item">
            <Link className="nav-link disabled" to="/">
              Disabled
            </Link>
          </li>
        </ul>
        <form className="form-inline mt-2 mt-md-0">
          <input
            className="form-control mr-sm-2"
            type="text"
            placeholder="Search"
            aria-label="Search"
          />
          <button
            className="btn btn-outline-success my-2 my-sm-0"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
    </div>
  </nav>
);

export default Header;
