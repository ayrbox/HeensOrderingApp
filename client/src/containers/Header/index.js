import Header from './connect';

export default Header;
