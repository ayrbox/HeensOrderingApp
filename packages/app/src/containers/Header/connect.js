import { connect } from 'react-redux';
import Header from './header';

// TODO: import action to open order modal (that will just toggle open)
// import

import { openOrderPane } from '../../actions/newOrderActions';

const mapState = state => ({
  testState: state.auth,
});

export default connect(mapState, {
  openOrderPane,
})(Header);
