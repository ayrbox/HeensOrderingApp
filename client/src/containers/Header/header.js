import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

import styles from './styles';

const Header = ({
  title,
  classes,
  openOrderModal,
}) => (
  <Fragment>
    <AppBar position="absolute" className={classes.root}>
      <Toolbar className={classes.toolbar}>
        <Typography
          variant="h5"
          color="inherit"
          noWrap
          onClick={() => console.log('terohead')}
        >
          {title}
        </Typography>
        <div className={classes.navItemsContainer}>
          <Button
            component={NavLink}
            className={classes.navItem}
            color="inherit"
            to="/orders"
          >
            Orders
          </Button>
          <Button
            component={NavLink}
            className={classes.navItem}
            color="inherit"
            to="/customers"
          >
            Customers
          </Button>
          <Button
            component={NavLink}
            className={classes.navItem}
            color="inherit"
            to="/menus"
          >
            Menus
          </Button>
          <Button
            component={NavLink}
            className={classes.navItem}
            color="inherit"
            to="/categories"
          >
            Categories
          </Button>
          <Button
            className={classes.navItem}
            color="inherit"
            onClick={openOrderModal}
          >
            + New Order
          </Button>
        </div>
      </Toolbar>
    </AppBar>
  </Fragment>
);

Header.propTypes = {
  title: PropTypes.string.isRequired,
  classes: PropTypes.shape().isRequired,
  openOrderModal: PropTypes.func.isRequired,
};

export default withStyles(styles)(Header);
