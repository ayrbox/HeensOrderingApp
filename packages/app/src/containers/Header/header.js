import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

import styles from './styles';

const Header = ({
  title,
  classes,
  openOrderPane,
}) => (
  <Fragment>
    <AppBar position="fixed" className={classes.root}>
      <Toolbar className={classes.toolbar}>
        <Typography
          variant="h5"
          color="inherit"
          noWrap
          data-testid="title"
        >
          {title}
        </Typography>
        <div className={classes.navItemsContainer}>
          <Button
            component={NavLink}
            className={classes.navItem}
            color="inherit"
            to="/orders"
          >
            Orders
          </Button>
          <Button
            component={NavLink}
            className={classes.navItem}
            color="inherit"
            to="/customers"
          >
            Customers
          </Button>
          <Button
            component={NavLink}
            className={classes.navItem}
            color="inherit"
            to="/menus"
          >
            Menus
          </Button>
          <Button
            component={NavLink}
            className={classes.navItem}
            color="inherit"
            to="/categories"
          >
            Categories
          </Button>
          <Button
            className={classes.navItem}
            color="inherit"
            onClick={openOrderPane}
            data-testid="new-order-button"
          >
            + New Order
          </Button>
        </div>
      </Toolbar>
    </AppBar>
  </Fragment>
);

Header.propTypes = {
  title: PropTypes.string.isRequired,
  classes: PropTypes.shape().isRequired,
  openOrderPane: PropTypes.func.isRequired,
};

export default withStyles(styles)(Header);
