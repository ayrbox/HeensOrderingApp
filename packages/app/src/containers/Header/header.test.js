import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Header from './header';

describe('<Header /> spec', () => {
  const mockOpen = jest.fn();
  let container;

  beforeEach(() => {
    container = render(
      <Router>
        <Header
          title="TestTitle"
          openOrderPane={mockOpen}
        />
      </Router>,
    );
  });

  afterEach(() => {
    mockOpen.mockClear();
    cleanup();
  });

  it('renders the component', () => {
    expect(container).toMatchSnapshot();
  });

  it('renders header with given title', () => {
    const { getByTestId } = container;
    const headerTitle = getByTestId('title');
    expect(headerTitle).toHaveTextContent(/TestTitle/);
  });

  it('should call open method on new order button click', () => {
    const { getByTestId } = container;
    fireEvent.click(getByTestId('new-order-button'));
    expect(mockOpen).toHaveBeenCalled();
  });
});
