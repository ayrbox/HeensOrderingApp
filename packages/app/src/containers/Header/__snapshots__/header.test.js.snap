// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root-15 MuiPaper-elevation4-21 MuiAppBar-root-6 MuiAppBar-positionFixed-7 MuiAppBar-colorPrimary-13 mui-fixed Header-root-1"
      >
        <div
          class="MuiToolbar-root-42 MuiToolbar-regular-44 MuiToolbar-gutters-43 Header-toolbar-2"
        >
          <h5
            class="MuiTypography-root-46 MuiTypography-h5-62 MuiTypography-colorInherit-75 MuiTypography-noWrap-72"
            data-testid="title"
          >
            TestTitle
          </h5>
          <div
            class="Header-navItemsContainer-3"
          >
            <a
              class="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-colorInherit-103 Header-navItem-4"
              href="/orders"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label-83"
              >
                Orders
              </span>
              <span
                class="MuiTouchRipple-root-111"
              />
            </a>
            <a
              class="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-colorInherit-103 Header-navItem-4"
              href="/customers"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label-83"
              >
                Customers
              </span>
              <span
                class="MuiTouchRipple-root-111"
              />
            </a>
            <a
              class="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-colorInherit-103 Header-navItem-4"
              href="/menus"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label-83"
              >
                Menus
              </span>
              <span
                class="MuiTouchRipple-root-111"
              />
            </a>
            <a
              class="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-colorInherit-103 Header-navItem-4"
              href="/categories"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label-83"
              >
                Categories
              </span>
              <span
                class="MuiTouchRipple-root-111"
              />
            </a>
            <button
              class="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-colorInherit-103 Header-navItem-4"
              data-testid="new-order-button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label-83"
              >
                + New Order
              </span>
              <span
                class="MuiTouchRipple-root-111"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root-15 MuiPaper-elevation4-21 MuiAppBar-root-6 MuiAppBar-positionFixed-7 MuiAppBar-colorPrimary-13 mui-fixed Header-root-1"
    >
      <div
        class="MuiToolbar-root-42 MuiToolbar-regular-44 MuiToolbar-gutters-43 Header-toolbar-2"
      >
        <h5
          class="MuiTypography-root-46 MuiTypography-h5-62 MuiTypography-colorInherit-75 MuiTypography-noWrap-72"
          data-testid="title"
        >
          TestTitle
        </h5>
        <div
          class="Header-navItemsContainer-3"
        >
          <a
            class="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-colorInherit-103 Header-navItem-4"
            href="/orders"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label-83"
            >
              Orders
            </span>
            <span
              class="MuiTouchRipple-root-111"
            />
          </a>
          <a
            class="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-colorInherit-103 Header-navItem-4"
            href="/customers"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label-83"
            >
              Customers
            </span>
            <span
              class="MuiTouchRipple-root-111"
            />
          </a>
          <a
            class="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-colorInherit-103 Header-navItem-4"
            href="/menus"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label-83"
            >
              Menus
            </span>
            <span
              class="MuiTouchRipple-root-111"
            />
          </a>
          <a
            class="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-colorInherit-103 Header-navItem-4"
            href="/categories"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label-83"
            >
              Categories
            </span>
            <span
              class="MuiTouchRipple-root-111"
            />
          </a>
          <button
            class="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-colorInherit-103 Header-navItem-4"
            data-testid="new-order-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-83"
            >
              + New Order
            </span>
            <span
              class="MuiTouchRipple-root-111"
            />
          </button>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
